sql/dd/impl/upgrade/dd.cc

```diff
@@ -1169,6 +1262,38 @@ bool upgrade_tables(THD *thd) {
                            (*it)->entity()->name().c_str());
   }
 
+  // myrocks has its own data dictionary
+  // after upgrade, fix these table name inside myrocks own data dictionary
+  if (bootstrap::DD_bootstrap_ctx::instance().is_dd_engine_change() &&
+      default_dd_storage_engine == DEFAULT_DD_ROCKSDB) {
+    handlerton *rocksdb_ddse = ha_resolve_by_legacy_type(thd, DB_TYPE_ROCKSDB);
+    if (!rocksdb_ddse) {
+      assert(false);
+      return true;
+    }
+    auto *handler =
+        get_new_handler(nullptr, false, thd->mem_root, rocksdb_ddse);
+    if (!handler) {
+      assert(false);
+      return true;
+    }
+
+    for (auto it = create_set.cbegin(); it != create_set.cend(); ++it) {
+      char lc_from[FN_REFLEN + 1];
+      char lc_to[FN_REFLEN + 1];
+      build_table_filename(lc_from, sizeof(lc_from) - 1,
+                           target_table_schema_name.c_str(), (*it).c_str(), "",
+                           0);
+      build_table_filename(lc_to, sizeof(lc_to) - 1, MYSQL_SCHEMA_NAME.str,
+                           (*it).c_str(), "", 0);
+      handler->ha_rename_table(lc_from, lc_to, nullptr, nullptr);
+    }
+    destroy(handler);
+
+    if (dd::execute_query(thd, "set global rocksdb_force_flush_memtable_now=1"))
```

Comment:
Why is this needed?

Reply:
For correctly, it isn't needed..it is easier to diagnostic if something goes wrong(check SST file is easier to walk through LSM). let me remove it first and may add it back in another diff.